replica/table.cc

```diff
@@ -1077,6 +1077,23 @@ table::sstable_list_builder::build_new_list(const sstables::sstable_set& current
     co_return make_lw_shared<sstables::sstable_set>(std::move(new_sstable_list));
 }
 
+future<>
+compaction_group::delete_sstables_atomically(std::vector<sstables::shared_sstable> sstables_to_remove) {
+    auto f = seastar::try_with_gate(_t._sstable_deletion_gate, [this, sstables_to_remove = std::move(sstables_to_remove)] {
+        return with_semaphore(_t._sstable_deletion_sem, 1, [sstables_to_remove = std::move(sstables_to_remove)] {
```

Comment:
need to make both lambdas `mutable` otherwise `std::move` will copy the const vector.